docs/docs/features/nats.md

```diff
@@ -0,0 +1,231 @@
+﻿---
+sidebar_position: 3
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import helloWorldModuleComponent from '!!raw-loader!./data/helloworld.module.yaml';
+import natsMessagesFormat from './data/ApiGearNatsv0.1.pdf';
+
+# NATS
+
+:::caution
+This is an experimental feature. It contains the smallest working set of functionalities to adapt the generated interface for use with the NATS protocol.
+It doesn't include any security, the error handling is minimal and it is not production ready. It doesn't use the JetStream nor the Streaming features. Please also check issues on github for this template.
+:::
+
+:::note
+A [CNATS](https://nats-io.github.io/nats.c/) library is used to implement NATS in this library.
+:::
+
+This feature provides NATS protocol adapters into your project, and also show that an existing protocol can be adapted for sharing your data in your ecosystem. When going through this document you may notice this implementation contains general client/server adapters in `📂hello-world/apigear/nats`
```

Comment:
"and also show that an existing protocol can be adapted for sharing your data in your ecosystem." Not sure what this means. protocol - adapted - data - ecosystem.

Reply:
It is- like with mqtt - to show that users can wite such feature by themself to adapt any existing protocol like whatever they use in their ecosystem, like it doesn't have to be olink which is not production proven. So this is example of how it can be done